### PR TITLE
Ruby 3 compatibility

### PR DESCRIPTION
**Pull request by tot-ioki**
This PR updates the dependencies in the gemspec and makes some minor adjustments to the code and specs to allow utilizing tickeos_b2b in newer Ruby/Rails versions (esp. ruby >= 3).

### CODE REVIEW

**Comment by tom-ioki** [comment]
I will do some testing and get back to you as soon as possible 👍

**Review by dal-ioki — APPROVED** [review]
LGTM, but I would wait for Tobi's review before merging :)

File: Gemfile.lock
Comment (tom-ioki) [inline Gemfile.lock:19]: This breaks the `basic_auth` in the client class. See this as reference: https://github.com/lostisland/faraday/issues/1317 and here's the solution (taken from the faraday documentation)

```ruby
Faraday.new(...) do |conn|
  conn.request :authorization, :basic, 'username', 'password'
end
```

**Review by tom-ioki — APPROVED** [review]

File: Gemfile.lock
Comment (tom-ioki) [reply Gemfile.lock:19]: Fixed it 👍

**Pull request merged**
